TypeSystem/flow-analysis/reachability_A12_t02.dart

```diff
@@ -11,16 +11,16 @@
 /// @description Checks that for expression of the form `E1 is S`
 /// `true(N) = promote(E1, S, after(E1))` and
 /// `false(N) = promote(E1, factor(T, S), after(E1))`. Test `factor(T, S)` for
-/// the case `T` is `R?` and `Null <: S`.
+/// the case `T` is `R?` and `S` is not subtype of `Null`.
 /// @author sgrekhov22@gmail.com
 
+import '../../Utils/static_type_helper.dart';
+
 main() {
   int? i = 2 > 1 ? null : 42;
   if (i is int) {
     int j = i;
```

Comment:
I'm afraid I was a little bit careless here. The current approach will succeed even if the static type of `i` after the test `i is int` happens to be `dynamic` or `Never`. So `i.expectStaticType<Exactly<int>>()` will be a stronger test than `int j = i;`.

Reply:
Thanks!